src/web/UserPresence.ts

```diff
@@ -6,40 +6,36 @@
 *
 * Web-specific implementation of the ReactXP interfaces related to
 * user presence.
+* 
+* User is considered present when user is focused on the App and has interacted with the App in the last 60 seconds.
+* User is considered not present, if app is not focused (backgrounded or blurred) or the app is focused 
+* but the user has not intereacted with the app in the last 60 seconds.
 */
 
 import RX = require('../common/Interfaces');
-
-// Hack to allow inline-require without importing node.d.ts
-declare var require: (path: string) => any;
-if (typeof(document) !== 'undefined') {
-    var ifvisible = require('ifvisible');
-}
+import AppVisibilityUtils from './utils/AppVisibilityUtils';
 
 export class UserPresence extends RX.UserPresence {
     private _isPresent: boolean;
 
     constructor() {
         super();
         // Handle test environment where document is not defined.
-        if (typeof(document) !== 'undefined') {
-            this._isPresent = ifvisible.now();
-
-            ifvisible.on('wakeup', this._handleWakeup.bind(this));
-            ifvisible.on('idle', this._handleIdle.bind(this));
-            ifvisible.on('focus', this._handleFocus.bind(this));
-            ifvisible.on('blur', this._handleBlur.bind(this));
-
-            window.addEventListener('blur', this._handleWindowBlur.bind(this));
+        if (typeof (document) !== 'undefined') {
```

Comment:
Nit: Space should be removed between "typeof" and "(document)".

Reply:
Removed.